pajbot/modules/chat_alerts/cheeralert.py

```diff
@@ -0,0 +1,277 @@
+            type="text",
+            required=True,
+            placeholder="{username} thank you so much for cheering {num_bits} bits! PogChamp",
+            default="",
+            constraints={"max_str_len": 400},
+        ),
+        ModuleSetting(
+            key="grant_points_per_100_bits",
+            label="Give points to user per 100 bits they cheer. 0 = off",
+            type="number",
+            required=True,
+            placeholder="",
+            default=0,
+            constraints={"min_value": 0, "max_value": 50000},
+        ),
+        ModuleSetting(
+            key="alert_message_points_given",
+            label="Message to announce points were given to user, leave empty to disable message. If the user cheers less than 100 bits, no message will be sent. | Available arguments: {username}, {points}, {num_bits}",
+            type="text",
+            required=True,
+            default="{username} was given {points} points for cheering {num_bits} bits! FeelsAmazingMan",
+            constraints={"max_str_len": 300},
+        ),
+    ]
+
+    def __init__(self, bot):
+        super().__init__(bot)
+
+    def on_cheer(self, user, num_bits):
+        """
+        A user just cheered bits.
+        Send the event to the websocket manager, and send a customized message in chat.
+        """
+
+        payload = {"username": user.name, "num_bits": num_bits}
+        self.bot.websocket_manager.emit("cheer", payload)
+
+        if self.settings["chat_message"]:
+            if num_bits >= 25000 and self.settings["twentyfivethousand_bits"] != "":
+                self.bot.say(self.get_phrase("twentyfivethousand_bits", **payload))
+            elif num_bits >= 10000 and self.settings["tenthousand_bits"] != "":
+                self.bot.say(self.get_phrase("tenthousand_bits", **payload))
+            elif num_bits >= 5000 and self.settings["fivethousand_bits"] != "":
+                self.bot.say(self.get_phrase("fivethousand_bits", **payload))
+            elif num_bits >= 1500 and self.settings["fifteenhundred_bits"] != "":
+                self.bot.say(self.get_phrase("fifteenhundred_bits", **payload))
+            elif num_bits >= 500 and self.settings["fivehundred_bits"] != "":
+                self.bot.say(self.get_phrase("fivehundred_bits", **payload))
+            elif num_bits == 420 and self.settings["fourtwenty_bits"] != "":
+                self.bot.say(self.get_phrase("fourtwenty_bits", **payload))
+            elif num_bits >= 100 and self.settings["hundred_bits"] != "":
+                self.bot.say(self.get_phrase("hundred_bits", **payload))
+            elif num_bits == 69 and self.settings["sixnine_bits"] != "":
+                self.bot.say(self.get_phrase("sixnine_bits", **payload))
+            elif self.settings["one_bit"] != "":
+                self.bot.say(self.get_phrase("one_bit", **payload))
+            else:
+                log.warning("Something went wrong with the bit amount announcement!!!!")
```

Comment:
This warning will trigger if someone donates 50 bits and no "one_bit" message is set, so shouldn't be needed anymore. I assume it served its purpose while testing though!